### DISCUSSION TITLE
How is .termux-build/LICENSE generated?

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-packages#16375] [Q&A] [answered]
Whenever I try to build a package, it always fails at installing the package license at this part:

https://github.com/termux/termux-packages/blob/757cc2510e87dfcaabe31debd4050cd6ae910621/scripts/build/termux_step_install_license.sh#L65

since apparently, the directory `~/.termux-build/LICENSE` does not exist (assuming it where `../../LICENSES/` points to). How do I manually add this licenses? Running `setup-ubuntu.sh` or `setup-archlinux.sh` is not possible since I'm using a exotic POSIX terminal emulator

**Comment by 2096779623** [comment]
Set `TERMUX_PKG_LICENSE_FILE` or `termux_step_install_license` in your `build.sh` .
https://github.com/termux/termux-packages/wiki/Creating-new-package#table-of-available-package-control-fields

**Comment by ghost** [comment, marked as answer]
After debugging for hours, I finally caught wind of the directory where all the LICENSES are which is https://github.com/termux/termux-packages/tree/master/packages/termux-licenses/LICENSES. I copied this directory to `~/.termux-build` and it's fixed.

**Reply by sylirre** [reply]
So basically you have a special version of `ln` which fails when source file does not exist?

`termux-licenses/LICENSES` is not supposed to be copied somewhere.

**Reply by ghost** [reply]
> So basically you have a special version of `ln` which fails when source file does not exist?

For context, I'm using msys2 in Windows 10 (msys envrionent) as a terminal so I think it's fair it would eventually break since this scripts wasn't designed to be run in a Windows machine. `ln` in msys2 requires that the source exists. It's kind of equivalent to `cp`.

I've only patched instances of `linux-x86_64` on the the scripts and replaced it `windows-x86_64` and also replaced the correct sha256sum when it's trying to download the termux-elf-cleaner https://github.com/harieamjari/termux-elf-cleaner/releases/tag/v2.1.1

So far, I've managed to compile simple packages like fftw, libpng, and zlib.
